src/jit/optimizer.cpp

```diff
@@ -3451,484 +3451,503 @@ bool Compiler::optComputeLoopRep(int        constInit,
-
-            /* if the HEAD is a BBJ_COND drop the condition (and make HEAD a BBJ_NONE block) */
-
-            if (head->bbJumpKind == BBJ_COND)
+            else
             {
-                phdr = head->bbTreeList;
-                noway_assert(phdr);
-                test = phdr->gtPrev;
-
-                noway_assert(test && (test->gtNext == nullptr));
-                noway_assert(test->gtOper == GT_STMT);
-                noway_assert(test->gtStmt.gtStmtExpr->gtOper == GT_JTRUE);
+                bbNew->bbJumpSwt  = bbIter->bbJumpSwt;
+                bbNew->bbJumpDest = bbIter->bbJumpDest;
+                bbNew->bbJumpOffs = bbIter->bbJumpOffs;
+            }
+        }
+    }
 
-                init = test->gtPrev;
-                noway_assert(init && (init->gtNext == test));
-                noway_assert(init->gtOper == GT_STMT);
+    // Unroll as outer loop expressions
+    for (unsigned int i = 0; i < outer; ++i)
+    {
+        BasicBlock* bbInsertAfter = bbBottom;
+        for (BasicBlock* bbIter = bbBody;; bbIter = bbIter->bbNext)
+        {
+            BasicBlock* bbNew = bbInsertAfter = fgNewBBafter(bbIter->bbJumpKind, bbInsertAfter, true);
+            if (!BasicBlock::CloneBlockState(this, bbNew, bbIter))
+            {
+                goto FAILED;
+            }
 
-                init->gtNext     = nullptr;
-                phdr->gtPrev     = init;
-                head->bbJumpKind = BBJ_NONE;
-                head->bbFlags &= ~BBF_NEEDS_GCPOLL;
+            if (bbIter == bbBottom)
+            {
+                bbNew->bbJumpKind = BBJ_NONE;
+                break;
             }
             else
             {
-                /* the loop must execute */
-                noway_assert(head->bbJumpKind == BBJ_NONE);
+                bbNew->bbJumpSwt  = bbIter->bbJumpSwt;
+                bbNew->bbJumpDest = bbIter->bbJumpDest;
+                bbNew->bbJumpOffs = bbIter->bbJumpOffs;
             }
+        }
+    }
 
-#ifdef DEBUG
-            if (verbose)
-            {
-                printf("Whole unrolled loop:\n");
+    if (lpIsFullUrl)
+    {
+        // We are doing full unrolling. we don't need any tests because loop is removed.
+        // check that they has any side-effects. if not, remove it.
+        GenTree* sideEffListTest = nullptr;
+        gtExtractSideEffList(gtTestExpr, &sideEffListTest, GTF_SIDE_EFFECT | GTF_ORDER_SIDEEFF);
 
-                gtDispTree(init);
-                printf("\n");
-                fgDumpTrees(head->bbNext, insertAfter);
-            }
-#endif
+        if (sideEffListTest)
+        {
+            gtTest->gtStmtExpr = sideEffListTest;
+            fgInsertStmtAtEnd(bbBottom, gtTest);
+        }
+        bbBottom->bbJumpKind = BBJ_NONE;
+    }
+    else
+    {
+        // We've removed gtTest from bbBottom for cloning. inserting it again.
+        fgInsertStmtAtEnd(bbBottom, gtTest);
+    }
 
-            /* Remember that something has changed */
+    if (lpIsPtclUrl)
```

Comment:
I think it seems need to add case when iteration operator is `GT_SUB.`
this will cause unpredictable behavior when the iterator operator is `GT_SUB`

maybe needs to modify gtInit for it.